### PR TITLE
Fix a shadowing setting in .golangci.yml

### LINKED ISSUE #5605: Update Golang linter configuration to avoid the deprecated `linters.govet.check-shadowing` option
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```
level=warning msg="[config_reader] The configuration option `linters.govet.check-shadowing` is deprecated. Please enable `shadow` instead, if you are not using `enable-all`."
level=error msg="[linters_context] typechecking error: named files must all be in one directory; have /github/workspace/response and /github/workspace/responsetranslator"
```

we haven't applied any linter configuration settings. we're wondering about that warning.
how we use it in our workflow:
```
      - name: Super Linter Code Base
        uses: github/super-linter/slim@v6
        env:
          FILTER_REGEX_EXCLUDE: "**/*_test.go"
          DEFAULT_BRANCH: master
          VALIDATE_GO_MODULES: true
          VALIDATE_BASH: true
          VALIDATE_ALL_CODEBASE: false
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Thanks

### Expected Behavior

no warning to appear.

### Super-Linter version

```markdown
slim image v6.4.1
```

### Relevant log output

```shell
see description.

https://github.com/centralnicgroup-opensource/rtldev-middleware-go-sdk/actions/runs/8939075947/job/24558831196?pr=136
```

### Steps To Reproduce

1) you may fork our go-sdk repo
2) we have a shareable workflow in place, related superlinter action plugin usage snippet  provided. go version installed v1.22.2
3) no linter config file in place for golang, so falling back to super-linter's defaults

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by massongit**
# Proposed changes

https://github.com/super-linter/super-linter/actions/runs/9095084838/job/24997762122

```
  level=warning msg=\"[config_reader] The configuration option `linters.govet.check-shadowing` is deprecated. Please enable `shadow` instead, if you are not using `enable-all`.\"
```

To resolve the above warning in go modules, I change `check-shadowing` in `.golangci.yaml` to enable `shadowing`.

Fix #5605

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

### CODE REVIEW

File: .github/linters/.golangci.yml [hunk 1]
```diff
@@ -31,8 +31,9 @@ linters-settings:
     # default is false: such cases aren't reported by default.
     check-blank: true
   govet:
-    # report about shadowed variables
```
Comment (ferrarimarco) [inline .github/linters/.golangci.yml:34]: We can delete this entire file because it seems to match with the default `TEMPLATES/.golangci.yml`, right?

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Thanks for this PR! A small thing to address.

File: .github/linters/.golangci.yml (hunk 1 above)
Comment (massongit) [reply .github/linters/.golangci.yml:34]: Thank you for your advice. I delete it: https://github.com/super-linter/super-linter/pull/5654/commits/71ad4303aa64546da268034d167280e413256247

**Review by zkoppert — APPROVED** [review]
✨

**Review by ferrarimarco — APPROVED** [review]

**Pull request merged**